packages/amplify-graphql-schema-generator/src/schema-generator/generate-schema.ts

```diff
@@ -1,8 +1,9 @@
 import { DirectiveWrapper, EnumWrapper, FieldWrapper, ObjectDefinitionWrapper } from '@aws-amplify/graphql-transformer-core';
-import { EnumValueDefinitionNode, Kind, print } from 'graphql';
+import { EnumValueDefinitionNode, Kind, print, parse, DocumentNode } from 'graphql';
 import { EnumType, Field, Index, Model, Schema } from '../schema-representation';
+import { applySchemaOverrides } from './schema-overrides';
 
-export const generateGraphQLSchema = (schema: Schema): string => {
+export const generateGraphQLSchema = (schema: Schema, existingSchema?: string): string => {
```

Comment:
Can we explicitly handle the invalid schema case? If `existingSchema` is invalid, throw a useful error message so that customer can fix it easily.

Reply:
Otherwise it would throw an error on parse statement on line 38 leaving no information to customer. Also, consider adding a unit test with invalid schema input.